packages/cadl-ranch-specs/http/type/model/inheritance/enum-discriminator/main.tsp

```diff
@@ -0,0 +1,50 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Illustrates inheritance with enum discriminator.")
+@scenarioService("/type/model/inheritance/enum-discriminator")
+namespace Type.Model.Inheritance.EnumDiscriminator;
+
+@doc("extensible enum type for discriminator")
+enum DogKind {
+  Golden: "golden",
+}
+
+@doc("Test extensible enum type for discriminator")
+@discriminator("kind")
+model Dog {
+  kind: DogKind;
+  weight: int32;
+}
+
+model Golden extends Dog {
+  kind: DogKind.Golden;
+}
+
+@scenario
+@scenarioDoc("""
+Receive model with extensible enum discriminator type.
+Expected response body:
+```json
+
+{"kind": "golden", "weight": 10}
+```
+""")
+@route("/extensible-enum")
+@get
+op getExtensibleEnum(): Dog;
```

Comment:
`getValid` or `getModel` be enough? `Dog` is not an enum.

Reply:
Yes. Previously I have test cases regarding fixed enum. But it looks like after `0.46.0` there will be error if you use fixed enum as discriminator type, which makes sense. So I deleted thec ases regarding fixed enum. I'll make a change.

Reply:
change is made

Reply:
> Was that an error or a warning? If it is a warning we should maybe just suppress it and add keep test for it. This is something I could see being allowed in very specific cases where all the discriminated types are know for sure

@timotheeguerin I double checked. It should be a warning originally. It's coverted as error in `pnpm validate-secenarios`. Two rules are triggered:
- https://github.com/Azure/typespec-azure/blob/0a95aee0a1d075b161ed2b8953578a81e9ebc611/packages/typespec-azure-core/src/lib.ts#L102C6-L102C33
- https://github.com/Azure/typespec-azure/blob/0a95aee0a1d075b161ed2b8953578a81e9ebc611/packages/typespec-azure-core/src/lib.ts#L121